File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def compare(x, y):
    s1 = x + y
    s2 = y + x
    
    if int(s1) > int(s2):
        return -1
    
    elif int(s1) < int(s2):
        return 1

    else:
        return 0
    
def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=cmp_to_key(compare))
    answer = str(int(''.join(numbers))) # 앞에 0이 나오는 경우 처리해야함
    return answer

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(int a, int b) {
    string sa = to_string(a); // 숫자를 문자열로 변환
    string sb = to_string(b); // 숫자를 문자열로 변환
    
    string n1 = sa + sb; // 문자열 합치기
    string n2 = sb + sa; // 문자열 합치기
    
    
    int num1 = stoi(n1);
    int num2 = stoi(n2);
    
    return num1 > num2;
}

string solution(vector<int> numbers) {
    sort(numbers.begin(), numbers.end(), compare);
    
    string answer = "";
    
    for (int i = 0; i < numbers.size(); i++) {
        answer += to_string(numbers[i]);
    }
    
    if (answer[0] == '0') {
        return "0";
    }
    
    else{
        return answer;
    }
}
